plugins/modules/change_request.py

```diff
@@ -242,7 +257,7 @@ def validate_params(params, change_request=None):
         )
 
 
-def ensure_present(module, table_client):
+def ensure_present(module, table_client, attachment_client):
```

Comment:
This method needs to validate and transform the `module.params["attachments"]` data before it uses it. The transformation should do the following transformation (I used YAML notation here):

```
- path: /p/a
- path: /p/b
  name: c
- path: /p/d
  type: e
- path: /p/f
  name: g
  type: h
````
should be transformed into:
```
a:
  path: /p/a
  type: <what mimetype detection returns>
  hash: <what sha256 returns>
c:
  path: /p/b
  type: <what mimetype detection returns>
  hash: <what sha256 returns>
d:
  path: /p/d
  type: e
  hash: <what sha256 returns>
f:
  path: /p/f
  type: h
  hash: <what sha256 returns>
```

If during the transformation we find a duplicated key, we should error out.

The transformed data structure is what we should then pass to other functions that operate on the attachments.

The transformation should go into module_utils/attachment.py since this transformation will be needed in all modules.